skrf/vi/vna/keysight/tests/test_fieldfox.py

```diff
@@ -0,0 +1,154 @@
+    ]
+)
+def test_params(
+    mocker, 
+    mocked_ff, 
+    param, 
+    expected_query, 
+    expected_write,
+    query_response, 
+    expected_val, 
+    write_val
+):
+    if expected_query is not None:
+        mocked_ff.query.return_value = query_response
+        test_query = getattr(mocked_ff, param)
+        mocked_ff.query.assert_called_once_with(expected_query)
+        assert test_query == expected_val
+
+    if expected_write is not None:
+        setattr(mocked_ff, param, write_val)
+        mocked_ff.write.assert_called_once_with(expected_write)
+
+def test_freq_query(mocker, mocked_ff):
+    mocked_ff.query.side_effect = [
+        '100', '200', '11'
+    ]
+    test = mocked_ff.frequency
+    assert test == skrf.Frequency(100, 200, 11, unit='hz')
+
+def test_freq_write(mocker, mocked_ff):
+    test_f = skrf.Frequency(100, 200, 11, unit='hz')
+    mocked_ff.frequency = test_f
+    calls = [
+        mocker.call("SENS:FREQ:STAR 100"),
+        mocker.call("SENS:FREQ:STOP 200"),
+        mocker.call("SENS:SWE:POIN 11"),
+    ]
+    mocked_ff.write.assert_has_calls(calls)
+
+def test_query_fmt_query(mocker, mocked_ff):
+    mocked_ff.query.side_effect = ['ASC,0', 'REAL,32', 'REAL,64']
+    test = mocked_ff.query_format
+    assert test == ValuesFormat.ASCII
+    test = mocked_ff.query_format
+    assert test == ValuesFormat.BINARY_32
+    test = mocked_ff.query_format
+    assert test == ValuesFormat.BINARY_64
+
+def test_query_fmt_write(mocker, mocked_ff):
+    mocked_ff.query_format = ValuesFormat.ASCII
+    mocked_ff.write.assert_called_with('FORM ASC,0')
+    mocked_ff.query_format = ValuesFormat.BINARY_32
+    mocked_ff.write.assert_called_with('FORM REAL,32')
+    mocked_ff.query_format = ValuesFormat.BINARY_64
+    mocked_ff.write.assert_called_with('FORM REAL,64')
+
+def test_define_msmnt(mocker, mocked_ff):
+    mocker.patch('skrf.vi.vna.keysight.FieldFox.ntraces', return_value=1, new_callable=mocker.PropertyMock)
+    mocked_ff.define_measurement(1, 'S11')
+    mocked_ff.write.assert_called_once_with('CALC:PAR1:DEF S11')
+
+# def test_calibration_write(mocker, mocked_ff):
+    # cal_terms = mocked_ff._cal_term_map
+    # mock_array = np.array([1+1j, 1+1j])
+    # cal_dict = {k: mock_array.copy() for k in cal_terms.keys()}
+    # mock_cal = skrf.Calibration.from_coefs(skrf.Frequency(100, 200, 11, unit='hz'), cal_dict)
+    # mocked_ff.calibration = mock_cal
+
+    # numpy defines == in a way that makes mocker.assert_has_calls not work
+    # Instead, we can just do the individual comparisons ourselves
+    # expected_calls = [
+        # mocker.call(f'SENS:CORR:COEF {term},', np.array([1.,1.,1.,1.]))
+        # for term in cal_terms.values()
+    # ]
+    # actual_calls = mocked_ff.write_values.call_args_list
+    # for actual, expected in zip(actual_calls, expected_calls):
+        # assert actual[0][0] == expected[1][0]
+        # np.testing.assert_array_almost_equal(actual[0][1], expected[1][1])
```

Comment:
to be kept?

Reply:
It might be able to be deleted. It was commented because I've not figured out how to do calibration yet. When implemented, it could be uncommented or added back in